photonlib-java-examples/src/main/java/org/photonlib/examples/simaimandrange/Robot.java

```diff
@@ -0,0 +1,123 @@
+import edu.wpi.first.wpilibj.PWMVictorSPX;
+import edu.wpi.first.wpilibj.TimedRobot;
+import edu.wpi.first.wpilibj.XboxController;
+import edu.wpi.first.wpilibj.controller.PIDController;
+import edu.wpi.first.wpilibj.drive.DifferentialDrive;
+import edu.wpi.first.wpilibj.util.Units;
+import org.photonlib.examples.simaimandrange.sim.DrivetrainSim;
+import org.photonvision.PhotonCamera;
+import org.photonvision.PhotonUtils;
+
+/**
+* The VM is configured to automatically run this class, and to call the functions corresponding to
+* each mode, as described in the TimedRobot documentation. If you change the name of this class or
+* the package after creating this project, you must also update the build.gradle file in the
+* project.
+*/
+public class Robot extends TimedRobot {
+    // 2020 High goal target height above ground
+    public static final double TARGET_HEIGHT_METERS = Units.inchesToMeters(81.19);
+
+    // Constants about how your camera is mounted to the robot
+    public static final double CAMERA_PITCH_RADIANS =
+            Units.degreesToRadians(15); // Angle "up" from horizontal
+    public static final double CAMERA_HEIGHT_METERS = Units.inchesToMeters(24); // Height above floor
+
+    // How far from the target we want to be
+    final double GOAL_RANGE_METERS = Units.feetToMeters(10);
+
+    // Change this to match the name of your camera
+    PhotonCamera camera = new PhotonCamera("photonvision");
+
+    // PID constants should be tuned per robot
+    final double LINEAR_P = 2.0;
+    final double LINEAR_D = 0.0;
+    PIDController forwardController = new PIDController(LINEAR_P, 0, LINEAR_D);
+
+    final double ANGULAR_P = 0.03;
+    final double ANGULAR_D = 0.003;
+    PIDController turnController = new PIDController(ANGULAR_P, 0, ANGULAR_D);
+
+    XboxController xboxController = new XboxController(0);
+
+    // Drive motors
+    PWMVictorSPX leftMotor = new PWMVictorSPX(0);
+    PWMVictorSPX rightMotor = new PWMVictorSPX(1);
+    DifferentialDrive drive = new DifferentialDrive(leftMotor, rightMotor);
+
+    @Override
+    public void teleopPeriodic() {
+        double forwardSpeed;
+        double rotationSpeed;
+
+        if (xboxController.getAButton()) {
+            // Vision-alignment mode
+            // Query the latest result from PhotonVision
+            var result = camera.getLatestResult();
+
+            if (result.hasTargets()) {
+                // First calculate range
+                double range =
+                        PhotonUtils.calculateDistanceToTargetMeters(
+                                CAMERA_HEIGHT_METERS,
+                                TARGET_HEIGHT_METERS,
+                                CAMERA_PITCH_RADIANS,
+                                Units.degreesToRadians(result.getBestTarget().getPitch()));
+
+                // Use this range as the measurement we give to the PID controller.
+                // -1.0 required to ensure positive PID controller effort _increases_ range
```

Comment:
Side note- this -1.0 (and a few others) probably need to be changed in the base aimandrange example - as-is, folks will have a few things reversed for the default drive base?

Reply:
Yeah that's likely true.